src/js/modules/wallet/controllers/sweep-coins-modal/sweep-coins-modal.ctrl.js

```diff
@@ -158,23 +187,47 @@
         $scope.publishRawTransaction = function () {
             var sdk = activeWallet.getSdkWallet().sdk;
 
-            sdk.sendRawTransaction($scope.sweepData.rawTx, function (err, result) {
-                $scope.working = false;
-
-                if(result.hash) {
-                    $timeout(function() {
-                        $scope.sweepData.txId = result.hash;
-                        $scope.form.step = $scope.STEPS.PUBLISH;
+            var transactions = $scope.sweepData.map(function (sweepTx) {
+                return sdk.sendRawTransaction(sweepTx.rawTx)
+                    .then(function (result) {
+                        sweepTx.txId = result.hash;
+                        return {
+                            status: true,
+                            sweep: sweepTx,
+                            txId: result.hash
+                        }
+                    }, function (err) {
+                        return {
+                            status: false,
+                            sweep: sweepTx,
+                            err: err
+                        }
                     });
-                    trackingService.trackEvent(trackingService.EVENTS.SWEEP.SWEEP_SUCCESS);
-                } else {
-                    trackingService.trackEvent(trackingService.EVENTS.SWEEP.SWEEP_FAIL);
-                    return dialogService.alert(
-                        $translate.instant("IMPORT_ERROR"),
-                        $translate.instant("TX_CANT_BE_PUSHED")
-                    ).result;
-                }
-            });
+                });
+
+                $q.all(transactions).then(function (results) {
+                    $scope.working = false;
+                    var failures = 0;
+                    results.map(function (result) {
+                        if(!result.status) {
+                            failures += 1;
+                        }
+                    });
+
+                    if (!failures) {
+                        $timeout(function () {
+                            // TODO: Display on PUBLISHED STATE MULTIPLE
+                            $scope.form.step = $scope.STEPS.PUBLISH;
+                        });
+                        trackingService.trackEvent(trackingService.EVENTS.SWEEP.SWEEP_SUCCESS);
+                    } else {
```

Comment:
This says if there are no failures show success, but we may have already broadcast 1 of the two transaction by then. It's due to an unlikely amount, but if we don't deal with this in an if, the else will catch it, which would say the it failed?

Reply:
"TX_CANT_BE_PUSHED_MULTIPLE": "{{ amount }} of the transactions can not be pushed. An error occurred."

Would be the new translation for one or more tx's failing.